### DISCUSSION TITLE
How to Contribute Support of new Domain?

### DISCUSSION THREAD

**Discussion by Johnson145** [OHF-Voice/intents#1464] [Supported Sentences]
Is there any way I can contribute supporting new domains?

I know that I can add custom intents to my own setup, but how would one add general support?

For instance, to add support for the _vacuum_ domain.

It's not enough to duplicate the `.yaml` files of an existing domain in this repository, right? Didn't find the code which is actually in charge of triggering the domain-specific business logic.

**Comment by tetele** [comment]
There isn't any.

As of the writing of this message, supported intents [are](https://github.com/home-assistant/intents/blob/main/intents.yaml):
- `HassTurnOn`
- `HassTurnOff`
- `HassGetState`
- `HassLightSet`

The only "hacks" are [properly treating](https://github.com/home-assistant/core/blob/dev/homeassistant/components/intent/__init__.py#L70) `HassTurnOn`/`HassTurnOff` for the `cover` domain so as not to add extra intents.

If you want to tailor these intents for the `vacuum` domain, there's nobody stopping you. If you want extra intents, though, [a discussion](https://github.com/home-assistant/intents/discussions) needs to happen first.
